packages/react/src/components/Table/TableHead/FilterHeaderRow/FilterHeaderRow.test.jsx

```diff
@@ -790,6 +791,51 @@ describe('FilterHeaderRow', () => {
     expect(screen.getByPlaceholderText('Choose an option')).toHaveValue('empty string');
   });
 
+  it('should display custom input when not undefined', () => {
+    render(
+      <FilterHeaderRow
+        {...commonFilterProps}
+        ordering={[{ columnId: 'col1' }, { columnId: 'col2' }]}
+        columns={[{ id: 'col1', customInput: () => <div>customInput</div> }, { id: 'col2' }]}
+      />
+    );
+
+    expect(screen.getAllByText('customInput')[0]).toBeInTheDocument();
+  });
+
+  it('should display custom input and filter data in the table', () => {
+    const Text = () => {
+      const [value, setValue] = React.useState('initial value');
+      return (
+        <TextInput
+          placeholder="Filter col1"
+          value={value}
+          onChange={(evt) => setValue(evt.target.value)}
+        />
+      );
+    };
+    render(
+      <FilterHeaderRow
+        {...commonFilterProps}
+        ordering={[{ columnId: 'col1' }, { columnId: 'col2' }]}
+        columns={[
+          {
+            id: 'col1',
+            customInput: ({ onChange }) => <Text onChange={onChange} />,
```

Comment:
why send the `onChange` function if this is not being used in the `Text` component you created at line 807?